torch/distributed/distributed_c10d.py

```diff
@@ -73,7 +76,20 @@ def __new__(cls, name):
                              "Gloo or MPI backend for collective operations "
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
-            raise ValueError("Invalid backend: '{}'".format(name))
+            try:
+                if name not in sys.modules:
+                    third_lib = importlib.import_module(name)
```

Comment:
This will import Python module with the same name as backend name. Just like example, if backend=gloo then it will try to import module "gloo". Such module names will be too abstract for PyTorch C10D-specific transport bindings. What is about to define the explicit rule for naming of such modules, for example "pytorch-<transport>" and try to import here "pytorch-" + name?

@mrshenli, @pietern, @apaszke  - what do you think?

Reply:
Good idea. I recommend improving the error checking here. It is possible that other modules with similar names exist and don't export the `ProcessGroupFoo` class. If that's the case, the backend will also be invalid, but likely results in some kind of `InvalidValue` error here instead.